doc/user_guide/user_guide.md

```diff
@@ -0,0 +1,281 @@
+
+The following SQL statement shows how to call an AAF query handler:
+
+```sql
+EXECUTE SCRIPT AAF_RUN_QUERY_HANDLER('{
+    "query_handler": {
+        "factory_class": {
+            "module": "<CLASS_MODULE>",
+            "name": "<CLASS_NAME>"
+        },
+        "parameter": "<CLASS_PARAMETERS>",
+        "udf": {
+            "schema": "<UDF_DB_SCHEMA>",
+            "name": "<UDF_NAME>"
+        }
+    },
+    "temporary_output": {
+        "bucketfs_location": {
+            "connection_name": "<BUCKETFS_CONNECTION_NAME>",
+            "directory": "<BUCKETFS_DIRECTORY>"
+        },
+        "schema_name": "<TEMP_DB_SCHEMA>"
+    }
+}');
+```
+
+See [Implementation of Custom Algorithms](#implementation-of-custom-algorithms) for a complete example.
+
+### Parameters
+
+| Parameter                    | Required? | Description                                                                   |
+|------------------------------|-----------|-------------------------------------------------------------------------------|
+| `<CLASS_NAME>`               | yes       | Name of the query handler class                                               |
+| `<CLASS_MODULE>`             | yes       | Module name of the query handler class                                        |
+| `<CLASS_PARAMETERS>`         | yes       | Parameters of the query handler class encoded as string                       |
+| `<UDF_NAME>`                 | -         | Name of Python UDF script that contains the algorithm implemented by the user |
+| `<UDF_DB_SCHEMA>`            | -         | Schema name where the UDF script is deployed                                  |
+| `<BUCKETFS_CONNECTION_NAME>` | yes       | BucketFS connection name which is used to create temporary bucketfs files     |
+| `<BUCKETFS_DIRECTORY>`       | yes       | Directory in BucketFS for the temporary bucketfs files                        |
+| `<TEMP_DB_SCHEMA>`           | yes       | Database Schema for temporary database objects, e.g. tables                   |
+
+Please take care to provide a string value for `<CLASS_PARAMETERS>`.  Simple data types like `float`, `int`, `bool` will be converted to a String, while a Json object or an array is represented as a string value with an unusable reference, e.g.  `table: 0x14823bd38580`.
+
+## Custom Algorithms
+
+### Deployment Options
+
+Using the AAF requires to implement a custom algorithm using one of the following alternatives
+* Adhoc implementation within a UDF
+* Build a custom extension
+
+#### Building a custom extension
+
+* Create a python package implementing the query handler.
+* Create an associated SLC which has the python package installed
+  * GitHub repository [python-extension-common](https://github.com/exasol/python-extension-common/) provides more detailed documentation and automation.
+* Leave out entry `udf` from the json input to use the default UDF. <!--
```

Comment:
Your version is better, you can remove the original comment

Reply:
Thanks :pray: